tests/test_state.py

```diff
@@ -1205,3 +1205,30 @@ def reset(self):
         err.value.args[0]
         == f"The event handler name `reset` shadows a builtin State method; use a different name instead"
     )
+
+
+@pytest.mark.asyncio
+async def test_state_with_invalid_yield():
```

Comment:
```suggestion
def test_state_with_invalid_yield():
```

only needs to be `async` if you are awaiting within the test.